R/importSNV.TCGA.R

```diff
@@ -0,0 +1,323 @@
+# importSNV.TCGA.R
+
+#' Import MAF files from source and converts them to rSNV files.
+#'
+#' \code{importSNV.TCGA} imports list of maf file names, creates rSNV or adds to
+#' existing rSNV file, return rSNV file
+#'
+#' @section MAF file:...
+#'
+#' @param fMAF vector of MAF file names
+#' @param rSNV rSNV file to merged new MAF data with
+#' @param na.rm remove all rows with NA values as attributes and drop from DF
+#' @param silent Controls whether output to console should be suppressed. FALSE
+#'   by default.
+#' @param writeLog Controls whether writing the result to the global logfile is
+#'   enabled. TRUE by default.
+#' @return rSNV file name containg data from all MAF files
+#'
+#' @family
+#'
+#'   ## @seealso \code{\link{importX}} importX() is used to import ...
+#'
+#'   ## @examples ## \dontrun{ ## importSNV.TCGA(IN, OUT) ## }
+#' @export
+
+importSNV.TCGA <- function(  fMAF,
+                             rSNV = NULL,
+                             na.rm = FALSE,
+                             silent = FALSE,
+                             writeLog = TRUE) {
+
+    myNotes <- character()
+
+    # ==== PARAMETERS ==========================================================
+
+    # NL <- .PlatformLineBreak()
+    # # Read header and one contents line
+    # tmp <- readLines(fName, n = 2)
+
+    # ==== VALIDATIONS =========================================================
+
+    # General parameter checks
+    cR <- character()
+    cR <- c(cR, .checkArgs(fMAF,         like = rep("FILE_E", length(fMAF)),
+                                                             checkSize = TRUE))
+    cR <- c(cR, .checkArgs(silent,       like = logical(1), checkSize = TRUE))
+    cR <- c(cR, .checkArgs(writeLog,     like = logical(1), checkSize = TRUE))
+    cR <- c(cR, .checkArgs(na.rm,        like = logical(1), checkSize = TRUE))
+
+    if(length(cR) > 0) {
+        stop(cR)
+    }
+
+    meta <- list(type = "rSNV",
+                 version = "1.0")
+
+    # ==== Validata fMAF: Extensions  =======================================
+    fileValidity <- function(file) {
+        if (strsplit(file, "\\.")[[1]][-1] %in% c("maf", "txt", "tsv")) {
+            return(TRUE)
+        } else {
+            return(FALSE)
+        }
+    }
+
+    mafHeaderCheck <- function(file) {
+        headValid <- TRUE
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,1] == "Hugo_Symbol") {
```

Comment:
Noooo. Don't define functions in here  :-)

Put the functions into the file separately, give them a function name that starts with a "." (e.g. .mafHeaderCheck) and remove the RoxygenHeader ( in particular the @export statement).